antarest/study/dao/link/storage_link_dao.py

```diff
@@ -0,0 +1,127 @@
+# Copyright (c) 2024, RTE (https://www.rte-france.com)
+#
+# See AUTHORS.txt
+#
+# This Source Code Form is subject to the terms of the Mozilla Public
+# License, v. 2.0. If a copy of the MPL was not distributed with this
+# file, You can obtain one at http://mozilla.org/MPL/2.0/.
+#
+# SPDX-License-Identifier: MPL-2.0
+#
+# This file is part of the Antares project.
+import typing as t
+
+from antares.study.version import StudyVersion
+
+from antarest.core.exceptions import LinkNotFound
+from antarest.study.business.model.link_model import LinkBaseDTO, LinkDTO, LinkInternal
+from antarest.study.business.utils import execute_or_add_commands
+from antarest.study.model import Study
+from antarest.study.storage.rawstudy.model.filesystem.factory import FileStudy
+from antarest.study.storage.storage_service import StudyStorageService
+from antarest.study.storage.variantstudy.model.command.create_link import CreateLink
+from antarest.study.storage.variantstudy.model.command.remove_link import RemoveLink
+from antarest.study.storage.variantstudy.model.command.update_link import UpdateLink
+
+
+class StorageLinkDao:
+    def __init__(self, storage_service: StudyStorageService):
+        self.storage_service = storage_service
+
+    def create_link(self, study: Study, link_dto: LinkDTO) -> LinkDTO:
```

Comment:
After more thought about the design of our "layers":
I think we should try to have business domain classes with plain names like `Link`, `Area`, etc.
The DAO will be in charge of implementing "CRUD" operations for those objects.
We should keep "DTO" naming for pure web layer objects.
However, it's something we can change later on, I think.